### DISCUSSION TITLE
SKGLView as MAUI deprecated OpenGLView Replacement

### DISCUSSION THREAD

**Discussion by mariomoramontero** [mono/SkiaSharp#2764] [General]
Hi,
As you may already know, The OpenGLView Control has been deprecated and completely removed from .NET8. We used it to display OpenGL video from our own made c++ video rendering library fully working on Android and MAUI .NET7, (and in Xamarin in the past). 

Now we need to find a proper replacement to do the same job. I thought that skiasharp SKGLView will fit perfectly and waited until the missing handlers issue was fixed recently on "3.0.0-preview.1.8" but It's not working, our app crashes. It's like our c++ library can't find the GL context, or using a different one, or there are any missing initialization. 
I counldn't find any sample on this direction, so I don't know if we are missing some initialization or we are completely mistaken to take this as an OpenGLView replacement at all...

As long as I know, we only have to :

1. call to .UseSkiaSharp() on builder to register handlers
2. Instantiate the SKGLView
3. Start the  rendering loop with "HasRenderLoop" then paint on the OnPaintSurface event handler. (Or calling InvalidateSurface instead of the render loop)

But it crashes all the time, so I'm wondering, from my lack of knowledge of GL rendering, if we are walking on the correct path or maybe missing something or if it's simply a bug.

thanks in advance
Mario

**Comment by taublast** [comment]
@mariomoramontero have you solved it with latest "3.0.0-preview.5.4"? What platform are you on?
